gpbft/gpbft.go

```diff
@@ -288,6 +288,20 @@ func (i *instance) receiveOne(msg *GMessage) error {
 		return nil
 	}
 
+	// Drop message that:
+	// * belong to future rounds, beyond the configured max lookahead threshold, and
+	// * carry no justification, i.e. are spammable.
+	//
+	// The only messages that are spammable are COMMIT for bottom. QUALITY and
+	// PREPARE messages may also not carry justification, but they are not spammable.
+	// Because, such messages are only valid at round 0; therefore, cannot belong
+	// to future rounds.
+	beyondMaxLookaheadRounds := msg.Vote.Round > i.round+i.participant.maxLookaheadRounds
+	spammable := msg.Vote.Step == COMMIT_PHASE && msg.Vote.Value.IsZero()
```

Comment:
Is is possible to change this predicate to `msg.Justification == nil && msg.Vote.Round > 0`? I feel it's just a little fragile in this form, and would rather that "only COMMIT bottom" is a description of the effect rather than the algorithm.